### PR TITLE
Publish scripts for npm package with types

### PR DESCRIPTION
**Pull request by tomasherceg**

### CODE REVIEW

File: ci/scripts/npm/build-dotvvm-types.ps1 [hunk 1]
```diff
@@ -1,5 +1,19 @@
...
+$typescriptVersion = $packageJson.devDependencies.typescript
+
+npm run tsc-types
+copy ./obj/typescript-types/dotvvm.d.ts "$dir/dotvvm-types/index.d.ts"
+
+pop-location
+
+push-location dotvvm-types
+
+npm install typescript@$typescriptVersion --save-dev
```
Comment (exyi) [inline ci/scripts/npm/build-dotvvm-types.ps1:16]: why do we need to install typescript in this package? The typescript compiler is executed in src/Framework/Framework, not here.

Comment (tomasherceg) [reply ci/scripts/npm/build-dotvvm-types.ps1:16]: It's only to put the same version of Typescript dependency into package.json here.

Comment (tomasherceg) [reply ci/scripts/npm/build-dotvvm-types.ps1:16]: Oh, I see. The dependency was there before, so I just updated the script to push there the same version that is used in DotVVM codebase. If it's not needed, we can safely remove it.

File: ci/scripts/npm/build-dotvvm-types.ps1 [hunk 2]
```diff
@@ -1,5 +1,18 @@
-push-location ../../DotVVM.Framework
+param([Parameter(Mandatory=$true)] $version)
 
-tsc --declaration --emitDeclarationOnly --skipLibCheck --outFile ../Tools/npm/dotvvm-types/index.d.ts
+$dir = pwd
+push-location ../../../src/Framework/Framework
+
+$packageJson = Get-Content package.json | ConvertFrom-Json
+$typescriptVersion = $packageJson.devDependencies.typescript
```
Comment (exyi) [inline ci/scripts/npm/build-dotvvm-types.ps1:7]: this is also not needed anymore

**Review by exyi — APPROVED** [review]

**Review by quigamdev — APPROVED** [review]
No big issue found. Just please resolve the comments.

**Pull request merged**